### PR TITLE
feat: Add F23 key support for Lenovo IdeaPad 5 pro and Thinkpad T14 / P14s

### PR DESCRIPTION
**Pull request by susnux**
The ThinkPad T14 and P14s (G5) as well as the IdeaPad 5 Pro 14AHP9 added support for the F23 key as part of the copilot key sequence.

The scan code is 0x6e following *PS/2 Set 1 Make*, so I guess this will also be valid for all other newer Lenovo atkbd driver based notebooks.

### CODE REVIEW

**Comment by poettering** [comment]
Why f23? Why not prog1?

**Comment by susnux** [comment]
> So the problem with this is that f23 means something on linux: it has been used for a long time to encode touchpad toggle, as a workaround for x11 keycode limitations.

TIL

On the other hand it is already "correctly" sent by some hardware, meaning mapping it to something different here will make the copilot key emit different sequences depending on your hardware and also different from what is documented to be sent by that key, which is probably unexpected behavior.
(Making it harder to remap this, see discussion leading to this PR: https://github.com/rvaiya/keyd/issues/825 ).

**Comment by poettering** [comment]
Yeah its a mess. Not sure what the best approach here is. @whot @jwrdegoede any idea what we should do with this new copilot key thing mess up? Apparently msft decided to ask hw manufacturers to map that new copilot key to some key combo with the scancode for f23... Which conflicts with the generally accepted interpretation of f23 on linux as touchpad toggle...

**Comment by poettering** [comment]
Maybe the solution is to simply accept the mess and inform desktop developers that F23 without modifiers should probably still be considered touchpad toggle, but if combined with shift+meta then it's the copilot button...

**Comment by whot** [comment]
Don't have the link to the specs at hand but MS decreed that the copilot key needs to Win+Shift+F23 instead of a dedicated key so we get to live with this decision for the next decade or so. Not altogether different to the Win+P -> display toggle in one of the previous decades.

On the other hand, I *think* this should be supportable OOTB once we get a new keysym added and add this to xkeyboard-config as part of the default keymap. I don't have all the moving parts in place for that yet though, but I've filed [xkeyboard-config#493](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/issues/493) now to have something to link against.

TLDR: systemd/udev shouldn't need to do anything here, once the XKB maps are in place this should work correctly.

**Comment by poettering** [comment]
is this hwdb patch necessary though? i.e. is explicitly mapping scancode 0x6e to F23 necessary, and lenovo specific? or is that something generic?

this appears to be entirely generic, i.e this nice PDF suggests any atkbd should use scancode 0x6e to F23:

https://download.microsoft.com/download/1/6/1/161ba512-40e2-4cc9-843a-923143f3456c/translate.pdf

I am a bit confused where the kernel has its mapping from ps/2 make 1 scancodes to keycodes, but if this is generic the F23 mapping really should be done there, instead of a hwdb mapping

**Comment by susnux** [comment]
> I am a bit confused where the kernel has its mapping from ps/2 make 1 scancodes to keycodes

Maybe here?
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/input/keyboard/atkbd.c?h=v6.12#n93
